contracts/src/v0.8/functions/dev/1_0_0/Functions.sol

```diff
@@ -0,0 +1,171 @@
+    Buffer.init(buffer.buf, DEFAULT_BUFFER_SIZE);
+
+    CBOR.writeString(buffer, "codeLocation");
+    CBOR.writeUInt256(buffer, uint256(self.codeLocation));
+
+    CBOR.writeString(buffer, "language");
+    CBOR.writeUInt256(buffer, uint256(self.language));
+
+    CBOR.writeString(buffer, "source");
+    CBOR.writeString(buffer, self.source);
+
+    if (self.args.length > 0) {
+      CBOR.writeString(buffer, "args");
+      CBOR.startArray(buffer);
+      for (uint256 i = 0; i < self.args.length; i++) {
+        CBOR.writeString(buffer, self.args[i]);
+      }
+      CBOR.endSequence(buffer);
+    }
+
+    if (self.encryptedSecretsReference.length > 0) {
+      if (self.secretsLocation == Location.Inline) {
+        revert NoInlineSecrets();
+      }
+      CBOR.writeString(buffer, "secretsLocation");
+      CBOR.writeUInt256(buffer, uint256(self.secretsLocation));
+      CBOR.writeString(buffer, "secrets");
+      CBOR.writeBytes(buffer, self.encryptedSecretsReference);
+    }
+
+    return buffer.buf.buf;
+  }
+
+  /**
+   * @notice Initializes a Chainlink Functions Request
+   * @dev Sets the codeLocation and code on the request
+   * @param self The uninitialized request
+   * @param codeLocation The user provided source code location
+   * @param language The programming language of the user code
+   * @param source The user provided source code or a url
+   */
+  function initializeRequest(
+    Request memory self,
+    Location codeLocation,
+    CodeLanguage language,
+    string memory source
+  ) internal pure {
+    if (bytes(source).length == 0) revert EmptySource();
+
+    self.codeLocation = codeLocation;
+    self.language = language;
+    self.source = source;
+  }
+
+  /**
+   * @notice Initializes a Chainlink Functions Request
+   * @dev Simplified version of initializeRequest for PoC
+   * @param self The uninitialized request
+   * @param javaScriptSource The user provided JS code (must not be empty)
+   */
+  function initializeRequestForInlineJavaScript(Request memory self, string memory javaScriptSource) internal pure {
+    initializeRequest(self, Location.Inline, CodeLanguage.JavaScript, javaScriptSource);
+  }
+
+  /**
+   * @notice Adds Remote user encrypted secrets to a Request
+   * @param self The initialized request
+   * @param encryptedSecretsReference Encrypted comma-separated string of URLs pointing to off-chain secrets
+   */
+  function addSecretsReference(Request memory self, bytes memory encryptedSecretsReference) internal pure {
+    if (encryptedSecretsReference.length == 0) revert EmptySecrets();
+
+    self.secretsLocation = Location.Remote;
+    self.encryptedSecretsReference = encryptedSecretsReference;
+  }
+
+  /**
+   * @notice Adds DON-hosted user slot id (referencing secrets) to a Request
+   * @param self The initialized request
+   * @param donSlotID Slot ID of the user's secrets hosted on DON
+   */
+  function addDONHostedSecrets(Request memory self, bytes1 donSlotID) internal pure {
```

Comment:
I had to change the function here https://github.com/smartcontractkit/chainlink/pull/9782
to use CBOR encoding.
Also I updated the comment in Request struct accordingly.